components/tree/private/item.jsx

```diff
@@ -39,32 +42,140 @@ const handleClick = (event, props) => {
 	}
 };
 
+const findNextNode = (flattenedNodes, node) => {
+	const nodes = flattenedNodes.map((flattenedNode) => flattenedNode.node);
+	const index = nodes.indexOf(node);
+	return flattenedNodes[(index + 1) % flattenedNodes.length];
+};
+
+const findPreviousNode = (flattenedNodes, node) => {
+	const nodes = flattenedNodes.map((flattenedNode) => flattenedNode.node);
+	let index = nodes.indexOf(node) - 1;
+	if (index < 0) {
+		index += flattenedNodes.length;
+	}
+	return flattenedNodes[index];
+};
+
+const handleKeyDownDown = (event, props) => {
+	if (props.focusedNodeIndex === props.treeIndex) {
+		// Select the next visible node
+		const flattenedNode = findNextNode(props.flattenedNodes, props.node);
+		props.onClick(
+			event,
+			{
+				node: flattenedNode.node,
+				select: true,
+				treeIndex: flattenedNode.treeIndex,
+			},
+			true
+		);
+	}
+};
+
+const handleKeyDownUp = (event, props) => {
+	if (props.focusedNodeIndex === props.treeIndex) {
+		// Go to the previous visible node
+		const flattenedNode = findPreviousNode(props.flattenedNodes, props.node);
+		props.onClick(
+			event,
+			{
+				node: flattenedNode.node,
+				select: true,
+				treeIndex: flattenedNode.treeIndex,
+			},
+			true
+		);
+	}
+};
+
+const handleKeyDownLeft = (event, props) => {
+	const nodes = props.flattenedNodes.map((flattenedNode) => flattenedNode.node);
+	const index = nodes.indexOf(props.parent);
```

Comment:
Whoops. This needs to be on the item not a branch. We need to close the parent on handleKeyDownLeft if on an item.

> Left arrow key on an end child node, *collapses the group* and moves :focus and aria-selected to the parent `treeitem`.

Reply:
I added this behavior in this commit.  And also added the tests!  https://github.com/salesforce/design-system-react/pull/1323/commits/1a36d887eecb89beaab50c543c49968bfd230799